### ISSUE TITLE
Issues with Payment Sheet not appearing

### ISSUE THREAD

**Issue by trex-quo** [stripe/stripe-react-native#315]
**Describe the bug**
Calling `initPaymentSheet()` and `presentPaymentSheet()` in immediate succession does not bring up the Payment Sheet. Adding a short sleep() statement (~1000ms) in between the two calls resolves the issue.  

Additionally, calling `presentPaymentSheet()` from a modal crashes the app. 

**To Reproduce**
```
  const onSubscribePress = async () => {
    let {error} = await initPaymentSheet({
      paymentIntentClientSecret: clientSecret,
      googlePay: true,
      merchantCountryCode: 'US',
      testEnv: true,
    });

    if (error) {
        //handle error
    }

    //without this line, the payment sheet will not open
    await sleep(1000);

    try {
      console.log('presenting payment sheet');
      ({error} = await presentPaymentSheet({
        clientSecret: clientSecret,
      }));
    } catch (e) {
      console.log({e});
      setLoading(false);
      return;
    }
  };
```

**Expected behavior**
It seems logical to choose not make a backend call until the client taps the payment button, meaning that the `init` and `present` call would fall under the same function. Sleeping is not an ideal solution to this problem, so it would be nice if the payment sheet was presented properly without having to do this. 

It would also be helpful if we could present the payment sheet from within a modal. As is, we need to redesign the app specifically for Android in order to present the payment sheet outside of our subscribe screen, which is contained in a modal. 

**Desktop (please complete the following information):**
 - OS: macOS Big Sur
 - Version 11.2.3

**Smartphone (please complete the following information):**
 - Device: iPhone 11, Galaxy S10
 - @stripe/stripe-react-native v0.1.4

**Comment by tayzlor** [comment]
Just a quick comment to mention I'm experiencing the same issue when you try and call both the initialise function and the present function off the back of a button press, and can confirm that a small delay does resolve the issue. 

Currently doing some further troubleshooting to understand why and will provide any more info when I have it!

**Comment by tayzlor** [comment]
So looks like this may be occurring because initPaymentSheet uses fragment commit() [https://developer.android.com/reference/android/app/FragmentTransaction.html#commit()] to update the paymentSheetFragment var in  https://github.com/stripe/stripe-react-native/blob/89042f4d54a7b525cea4a245ce8a7fc8ebf8de03/android/src/main/java/com/reactnativestripesdk/StripeSdkModule.kt#L273-L275

The problem with commit() is that it just schedules the commit() but the commit does not happen immediately. 

Android API 24 added a commitnow() function which runs the commit synchronously. https://developer.android.com/reference/android/app/FragmentTransaction.html#commitNow() 

Possible solution to the above would be to use commitnow() during the initPaymentSheet method?

**Comment by thorsten-stripe** [comment]
Thanks for investigating, we'll take a look.

**Comment by manish-shukla01** [comment]
i ran into the same issue. although even with sleep for 2 seconds, issue did not disappear. i tried only with test card. need to try with real card, not sure if it will be any different with real card. i can try to change the UI little bit so these action dont happen right after one another but is this problem going to be fixed. if yes, i can leave as is.

**Comment by jalbertsr** [comment]
This same issue keeps happening to me, and I'm following exactly as it is explained in your [docs](https://stripe.com/docs/payments/accept-a-payment?platform=react-native&ui=payment-sheet-custom). I was using version 1.4 and I did saw this issue was apparently solved in 1.5, so I update to 2.0.0. But it keeps happening :(

I'm using the `StripeProvider` wrapping my component, inside my `useEffect` you can find the `initializePaymentSheet` that fetches the `customerId`, `customerEphemeralKeySecret` and `paymentIntentClientSecret` and just after having those I call `initPaymentSheet`. Only after the user presses the button to pay, I do call `presentPaymentSheet`, it doesn't matter the time I wait, it always ends with the same (not very meaningful) error.

are we sure this is 100% fixed @thorsten-stripe ? any hints? 

thanks in advance.

**Comment by schoenbl** [comment]
Same thing is happening to me. OP's solution works as well.
